clipdel

```diff
@@ -16,72 +16,164 @@ cache_file_prefix=$cache_dir/line_cache
+            return 2
+        fi
+    else
+       printf '%s\n' "${3:-Options '$1' and '$CM_FUNCTION' conflict}" >&2
+       return 1
+    fi
+}
 
-".*" is special, it will just nuke the entire data directory, including the
-line caches and all other state.
+print_usage() {
+    cat << 'EOF'
+Usage: clipdel [OPTION] [REGEX]
+Delete clipmenu entries.
 
 Arguments:
 
-    -d  Delete for real.
+    --all                      Delete all entries
+    -c, --current              Delete current entry
+    -d, --delete               Delete for real (default when deleting all, current or from stdin)
+    -t, --older-than <STRING>  Delete entries older than STRING (format: N UNIT [N UNIT...])
+    -h, --help                 Print this message
+
+Examples:
+
+    clipmenu | clipdel         Launch clipmenu and delete selected entry
+    clipdel -d "bye world"     Delete all entries containing "bye world"
+    clipdel -t "10 hours" -d   Delete all entries older than 10 hours
+    clipdel -t "1 year 1 day"  List entries older than 1 year plus 1 day
 
 Environment variables:
 
 - $CM_DIR: specify the base directory to store the cache dir in (default: $XDG_RUNTIME_DIR, $TMPDIR, or /tmp)
 EOF
-    exit 0
-fi
-
-line_cache_files=( "$cache_file_prefix"_* )
+}
+
+while (($# > 0)); do
+    case "$1" in
+        --all|-c|--current) check_conflict "$1" 0 || exit 1;;
+        -d|--delete) CM_REAL_DELETE=1;;
+        -h|--help)
+            print_usage
+            exit 0
+            ;;
+        -t|--older-than)
+            check_conflict "$1" 0 || exit 1
+            shift
+            if [[ "$1" ]]; then
+                date_format=$(sed "s,\([0-9]\+ [A-Za-z]\+\),\1 ago,g" <<< "$1")
+                CM_OLDER_THAN=$(date --date="$date_format" +%s%N 2> /dev/null) ||
+                {
+                    printf '%s\n' "Unrecognized date format '$1'" >&2
+                    exit 1
+                }
+            else
+                printf '%s\n' "Missing date format" >&2
+                exit 1
+            fi
+            ;;
+        --|[!-]*)
+            if ((CM_REQUIRE_PATTERN)) && ! ((${#CM_PATTERNS[@]})); then
+                if [[ "$1" == "--" ]]; then
+                    shift
+                    (( $# )) && CM_PATTERNS=("$*")
+                    break
+                fi
+                CM_PATTERNS=("$1")
+            else
+                printf '%s\n' "Too many arguments" >&2
+                exit 1
+            fi
+            ;;
+        -*)
+            printf '%s\n' "Unrecognized option '$1'" >&2
+            exit 1
+            ;;
+    esac
+    shift
+done
 
 if (( ${#line_cache_files[@]} == 0 )); then
     printf '%s\n' "No line cache files found, no clips exist" >&2
     exit 0  # Well, this is a kind of success...
-fi
-
-# https://github.com/koalaman/shellcheck/issues/1141
-# shellcheck disable=SC2124
-raw_pattern=$1
-esc_pattern=${raw_pattern//\#/'\#'}
-
-if ! [[ $raw_pattern ]]; then
+elif [[ -p /dev/stdin ]]; then
```

Comment:
stdin doesn't always have to be a pipe, even if being read, so I don't think this will work:

```
$ cat s
#!/bin/bash
[[ -p /dev/stdin ]]
echo $?
$ ./s
1
$ ./s < s
1
```

Reply:
It should now work with `./s < s` and `./s <<< "string"`